### ISSUE TITLE
MaxTo crashes when viewing regions and exiting "cancel changes"

### ISSUE THREAD

**Issue by RhinoFox** [digitalcreations/MaxTo#635]
First of all, after installing 2.1.0. Stable over 2.1 beta 5. My own region are vanished.

When I look at te regions, press on presets. I don't see, my own.
Clicking on [Cancel changes] make MaxTo crash.
Restart MaxTo.Core, says there was a crash.
Starting again tells me MAxTo is already runnig. I don't see anything in the notification area, nor in the task manager, but procesexplorer does.

[maxto20201030.log](https://github.com/digitalcreations/MaxTo/files/5464984/maxto20201030.log)

**Comment by vegardlarsen** [comment]
Thank you for reporting this. This seems to be the stack trace that caused your issue:

```
2020-10-30 12:55:56 [UserInterface@2.1.0.697] [Fatal] [] Unhandled exception "TargetInvocationException" at "   bij System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   bij System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   bij System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   bij Caliburn.Micro.XamlPlatformProvider.<>c__DisplayClass13_3.<GetViewCloseAction>b__1()
   bij Caliburn.Micro.XamlPlatformProvider.OnUIThread(Action action)
   bij Caliburn.Micro.Screen.TryClose(Nullable`1 dialogResult)
   bij MaxTo.UI.ViewModels.SetRegionsWindowManager.<Activate>b__11_1(Object sender, DeactivationEventArgs e)
   bij System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   bij Caliburn.Micro.Screen.Caliburn.Micro.IDeactivate.Deactivate(Boolean close)
   bij Caliburn.Micro.WindowManager.WindowConductor.Closed(Object sender, EventArgs e)
   bij System.EventHandler.Invoke(Object sender, EventArgs e)
   bij System.Windows.Window.OnClosed(EventArgs e)
   bij System.Windows.Window.WmDestroy()
   bij System.Windows.Window.WindowFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   bij System.Windows.Interop.HwndSource.PublicHooksFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   bij MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   bij MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   bij System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   bij System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   bij System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   bij MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)"
System.Reflection.TargetInvocationException: Het doel van een aanroep heeft een uitzondering veroorzaakt. ---> System.InvalidOperationException: Kan Visibility niet instellen op Visible, of Show, ShowDialog, Close of WindowInteropHelper.EnsureHandle aanroepen wanneer een venster wordt gesloten.
   bij System.Windows.Window.VerifyNotClosing()
   bij System.Windows.Window.InternalClose(Boolean shutdown, Boolean ignoreCancel)
   bij System.Windows.Window.Close()
   --- Einde van intern uitzonderingsstackpad ---
   bij System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   bij System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   bij System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   bij Caliburn.Micro.XamlPlatformProvider.<>c__DisplayClass13_3.<GetViewCloseAction>b__1()
   bij Caliburn.Micro.XamlPlatformProvider.OnUIThread(Action action)
   bij Caliburn.Micro.Screen.TryClose(Nullable`1 dialogResult)
   bij MaxTo.UI.ViewModels.SetRegionsWindowManager.<Activate>b__11_1(Object sender, DeactivationEventArgs e)
   bij System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   bij Caliburn.Micro.Screen.Caliburn.Micro.IDeactivate.Deactivate(Boolean close)
   bij Caliburn.Micro.WindowManager.WindowConductor.Closed(Object sender, EventArgs e)
   bij System.EventHandler.Invoke(Object sender, EventArgs e)
   bij System.Windows.Window.OnClosed(EventArgs e)
   bij System.Windows.Window.WmDestroy()
   bij System.Windows.Window.WindowFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   bij System.Windows.Interop.HwndSource.PublicHooksFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   bij MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   bij MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   bij System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   bij System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   bij System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   bij MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
```

Is this reproducible consistently on your machine? What are the exact steps you take to make it crash like this?

**Comment by RhinoFox** [comment]
Right-click  on MacTo-icon in the notification area > Change Regions > Load preset > Cancel Changes
[maxto20201030.log](https://github.com/digitalcreations/MaxTo/files/5466239/maxto20201030.log)

BTW: Where should I find the stack trace. It could be this. There's Dutch tekst in it.

**Comment by RhinoFox** [comment]
After restoring my last config.json (before upgrade). Problem solved

**Comment by vegardlarsen** [comment]
Thank you for reporting this. It was something that only happens on multi-monitor solutions, and since Covid I've been on a single-monitor system, which is why I didn't catch this. I dug out an old monitor and was able to reproduce it.

The fix will be in 2.1.1.
